plugin/storage/grpc/shared/grpc_client.go

```diff
@@ -29,9 +29,20 @@ import (
 
 // grpcClient implements shared.StoragePlugin and reads/writes spans and dependencies
 type grpcClient struct {
-	readerClient     storage_v1.SpanReaderPluginClient
-	writerClient     storage_v1.SpanWriterPluginClient
-	depsReaderClient storage_v1.DependenciesReaderPluginClient
+	allowTokenFromContext bool
```

Comment:
is the config option actually required? If the query-service is configured to pass the bearer token, it would include it in the context, and this code just needs to respect that.

Reply:
We do want a way to tell our components that the token shouldn't be propagated.

Reply:
There is already configuration to that effect in the query service, which controls token handoff from query to storage. I don't think yet another flag for storage-storagePlugin is needed.